lib/pbench/cli/agent/commands/tools/clear.py

```diff
@@ -187,10 +175,17 @@ def is_empty(path):
         return not any(path.iterdir())
 
 
+def contains_empty(items: str) -> bool:
+    """Determine if the comma separated string contains en empty element"""
+    return bool([name for name in items.split(",") if not name])
+
+
 def _group_option(f):
     """Group name option"""
 
     def callback(ctxt, _param, value):
+        if value and contains_empty(value):
```

Comment:
The callback gets activated if the `value` is falsey? Or, if we get here from something like `--group=''` shouldn't we fail with an empty group anyway?

Reply:
That's an interesting point, yeah I guess we should fail if someone specifies option like `--group=''` instead of assuming this means `default` to avoid unexpected results. `default` should only be assumed in case if no `group` option is specified. Similarly, for `remotes` and `tool_names` en empty value should not be assumed as not specifying an option.

Reply:
If you want to be all orthodox and pedantic, sure.  😁  Although, personally, I have no problem with allowing people to specify a blank value in order to explicitly request the default (e.g., to override a previous specification...but that's probably a discussion for a different forum...).

Reply:
Right, but just a thought, what if someone does `-group=${g1}` and unexpectedly `g1` happens to be empty, then by the end we would have deleted user's default group.

Reply:
Hmm. so empty group is OK, if it's the only group, but not if it has friends? Then why isn't `--group=,tool` a request for the default group and the group `tool`??

Yeah, no; I don't want to have this discussion either. It strikes me as inconsistent with the direction we've pushed this; but my main question was whether click really even sends us here with an empty value. `pbench-clear-tools --group` gets rejected because it requires a value. In the *current* agent, `pbench-clear-tools --group=''` gets rejected as "invalid --group option". Do we want to mess with that here?

Reply:
> Right, but just a thought, what if someone does `-group=${g1}` and unexpectedly `g1` happens to be empty, then by the end we would have deleted user's default group.

Yeah, I'd rather not implicitly mess with the default group when someone says `--group`. Maybe `--group=default` is OK (does that work now?), but I don't think we should, um, default to default in this particular scenario.

Reply:
So Dave, yes, we do fail with the current setup if someone specifies  `--group=''`  click does that for us. And this works `--group=default` it correctly removes the default group

Reply:
> what if someone does `--group=${g1}`

I would defer to Click on that.  (That is, I could imagine that `--group=${g1}` with `g1=''` might produce a different result from `--group=''`, depending on when and how the substitution happens and the quotes get removed, respectively.)

> so empty group is OK, if it's the only group, but not if it has friends?

Yeah, I can see how that might seem inconsistent...I'd have to think about how to explain how it's not.  :wink:

> Maybe `--group=default` is OK

The way the code is currently structured, there really isn't a _default group_...rather, there's a group called `"default"`, which is just like any other group, except that, if you invoke a command and don't explicitly specify a group, that's the one you get (oh, and, yeah, we do this odd thing where, once it's created, we never delete its directory...).  And, yes, you can specify it explicitly.  But, I'm not inclined to mess with any of that now.

I don't know whether our code can tell the difference between `--group` being omitted, it being specified without a value, and it being specified with a blank value (in one of several ways).  If it can, then I'm OK with us using `"default"` when the switch or it's value is omitted, and I'm willing to go along if we want to flag an error for an explicitly-specified empty value.

Reply:
First; yes, with click we can tell the difference between `--group` and `--group=''`. The former is a *click* error because the argument is required; the latter is a Pbench error because "" isn't a group. (And if "someone" actually created a `/var/lib/pbench-agent/tools-v1-` directory, just hypothetically, the (old) `pbench-clear-tools` will empty the directory and be happy as a clam that's happy... 🙈 )

```
[root@dhcp31-136 ~]# pbench-clear-tools --group
Error: Option '--group' requires an argument.
[root@dhcp31-136 ~]# pbench-clear-tools --group=''
	pbench-clear-tools: invalid --group option (""), Bad tool group, '': directory /var/lib/pbench-agent/tools-v1- does not exist
Usage: pbench-clear-tools [OPTIONS]

  clear all tools or filter by name or group
```

Reply:
But, OK, this issue isn't resolved. We're not doing anything to fail "cleanly and proactively" on `--group=""` so I imagine the new code will fail much like the old code with missing directory `/var/lib/pbench-agent/tools-v1-`. I'm not entirely OK with that, but I suppose I won't pursue the issue in this context so I'll resolve the *conversation*. ☹️